File: app/src/main/java/org/mesonet/app/forecast/ForecastModel.java
package org.mesonet.app.forecast;


import org.mesonet.app.formulas.DefaultUnits;
import org.mesonet.app.formulas.UnitConverter;

public class ForecastModel implements DefaultUnits
{
    @Override
    public UnitConverter.TempUnits GetDefaultTempUnit() {
        return UnitConverter.TempUnits.kFahrenheit;
    }

    @Override
    public UnitConverter.LengthUnits GetDefaultLengthUnit() {
        return null;
    }

    @Override
    public UnitConverter.SpeedUnits GetDefaultSpeedUnit() {
        return UnitConverter.SpeedUnits.kMps;
    }

    @Override
    public UnitConverter.PressureUnits GetDefaultPressureUnit() {
        return null;
    }

    public enum HighOrLow { High, Low }
    public String mTime;
    public String mIconUrl;
    public String mStatus;
    public HighOrLow mHighOrLow;
    public Number mTemp;
    public UnitConverter.CompassDirections mWindDirection;
    public UnitConverter.CompassDirections mWindGustsDirection;
    public Number mWindSpd;
    public Number mWindGusts;
}

File: app/src/main/java/org/mesonet/app/forecast/ForecastModel.kt
package org.mesonet.app.forecast


import org.mesonet.app.formulas.DefaultUnits
import org.mesonet.app.formulas.UnitConverter

class ForecastModel : DefaultUnits {
    var mTime: String? = null
    var mIconUrl: String? = null
    var mStatus: String? = null
    var mHighOrLow: HighOrLow? = null
    var mTemp: Number? = null
    var mWindDirection: UnitConverter.CompassDirections? = null
    var mWindGustsDirection: UnitConverter.CompassDirections? = null
    var mWindSpd: Number? = null
    var mWindGusts: Number? = null
    override fun GetDefaultTempUnit(): UnitConverter.TempUnits {
        return UnitConverter.TempUnits.kFahrenheit
    }

    override fun GetDefaultLengthUnit(): UnitConverter.LengthUnits? {
        return null
    }

    override fun GetDefaultSpeedUnit(): UnitConverter.SpeedUnits {
        return UnitConverter.SpeedUnits.kMps
    }

    override fun GetDefaultPressureUnit(): UnitConverter.PressureUnits? {
        return null
    }

    enum class HighOrLow {
        High, Low
    }
}
